elasticapm/transport/base.py

```diff
@@ -135,7 +175,22 @@ def close(self):
         Cleans up resources and closes connection
         :return:
         """
-        self.flush(sync=True, start_flush_timer=False)
+        if self._closed:
+            return
+        self._closed = True
+        self.queue("close", None)
+        if not self._flushed.wait(timeout=self._max_flush_time):
+            raise ValueError("close timed out")
+
+    def flush(self):
+        """
+        Trigger a flush of the queue.
+        Note: this method will only return once the queue is empty. This means it can block indefinitely if more events
```

Comment:
I know this is only meant for tests, but FWIW: in the Go agent, flush just flushes the _currently_ enqueued events. When  a flush command is received by the background thread, it gets the current size of the queue and drains it by that many events. Even that's not necessary here because you have a single queue (whereas in Go we have one per event type, and a separate event/condition variable for flushing.)